src/hypertable.c

```diff
@@ -1677,7 +1723,10 @@ ts_hypertable_create_from_info(Oid table_relid, int32 hypertable_id, uint32 flag
 	 * lock upgrades, which are susceptible to deadlocks. If we aren't
 	 * migrating data, then shouldn't have much contention on the table thus
 	 * not worth optimizing.
+	 * Also obtaining locks on referenced tables through foreign keys to avoid
+	 * deadlocks, see issue 1652.
 	 */
+	ts_hypertable_get_and_lock_referenced_tables(table_relid, AccessExclusiveLock);
```

Comment:
I strongly think we should not try to fix the potential deadlock issue. Deadlocks occasionally occur in databases. PG tried to fix this by terminating one of the transactions (the segfault should be fixed and the other PR probably does it). Here are the reasons why it is not worth trying to fix this: 1) the fix is not fool proof. We cannot control the order in which the FK table locks were acquired by the other transaction. The order of acquisition matters when we want to prevent  a deadlock.2) create_hypertable is an infrequent operation. (unlike drop_chunks which is a more frequent operation and that fix is more of a bandage for reasons stated in 1 ). We should just let PG's deadlock avoidance mechanism handle this. The user should retry the operation.